Muon miniAOD DQM

sscruz opened the pull request.

muffinfan committed:
New MillePede production system.

 - less error-prone
 - unified track selection and refitting sequence
 - partial rewrite in python of old perl scripts
 - old production system left operational for the time being

cmsbuild commented:
A new Pull Request was created by  @sscruz  for CMSSW_8_0_X.

It involves the following packages:

DQMOffline/Configuration
DQMOffline/Muon


@cmsbuild, @vanbesien, @deguio, @davidlange6 can you please review it and eventually sign? Thanks.
@barvic, @battibass, @threus, @abbiendi, @jhgoh, @bellan, @trocino, @rociovilar this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

Following commands in first line of a comment are recognized

* **+1|approve[d]|sign[ed]**: L1/L2's to approve it
* **-1|reject[ed]**: L1/L2's to reject it
* **assign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to request signatures from other categories
* **unassign &lt;category&gt;[,&lt;category&gt;[,...]]**: L1/L2's to remove signatures from other categories
* **hold**: L1/all L2's/release manager to mark it as on hold
* **unhold**: L1/user who put this PR on hold
* **merge**: L1/release managers to merge this request
* **[@cmsbuild,] please test**: L1/L2 and selected users to start jenkins tests
* **[@cmsbuild,] please test with cms-sw/cmsdist#&lt;PR&gt;**: L1/L2 and selected users to start jenkins tests using externals from cmsdist

deguio commented:
this is apparently still not mergeable.

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

cmsbuild commented:
-1 
Tested at: 1c5d15360df1ef03145afe7c68722b9165ed24f0
 I found an error when building: 
 
 <pre>>> Pluging of all type refreshed.
gmake[1]: Target `PostBuild' not remade because of errors.
gmake[1]: Leaving directory `/tmp/cmsbuild/workspace/ib-any-integration/CMSSW_8_0_X_2016-01-20-2300'
gmake: *** [src] Error 2
gmake: Target `all' not remade because of errors.
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10649/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b32b4ead0acbc64f84c194e4db21945dc2328458
https://github.com/cms-sw/cmssw/commit/4dc6357c043a05e5f3e6816305bc65465f723533
https://github.com/cms-sw/cmssw/commit/982e9a7dd5952c54b9a453882649f5899d37e3b2
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10649/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10649/git-merge-result

sscruz commented:
The compilation errors were fixed in the last commit.

deguio commented:
please test
I am reviewing the code and I have comments. I will post them when I am done with the review.
cheers,
F.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: c8b1236b1a4d6ebbc4b660abf3ed4f5f3a77cf7f
 When I ran the RelVals I found an error in the following worklfows: 
 1001.0 step1
<pre>DAS Error</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10652/summary.html 

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b32b4ead0acbc64f84c194e4db21945dc2328458
https://github.com/cms-sw/cmssw/commit/4dc6357c043a05e5f3e6816305bc65465f723533
https://github.com/cms-sw/cmssw/commit/982e9a7dd5952c54b9a453882649f5899d37e3b2
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10652/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10652/git-merge-result

folguera commented:
the log of the error found is not very informative, I believe it was due to some miscommunication with DAS, could you please have a look and confirm?

deguio commented:
please test
das error not related.

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented on DQMOffline/Configuration/python/DQMOffline_SecondStep_cff.py:
```diff
@@ -112,15 +112,17 @@
                                cscOfflineCollisionsClients *
                                muonQualityTests
                                )
+DQMHarvestMuon_miniAOD = cms.Sequence( muonQualityTests_miniAOD )
```
this is not needed. please remove.

deguio commented on DQMOffline/Configuration/python/DQMOffline_cff.py:
```diff
@@ -137,3 +138,12 @@
+
+from DQMOffline.Muon.miniAOD_cff import *                                                                 
+
+muonMiniAOD = cms.Sequence( MuonMiniAOD )
```
not needed

deguio commented on DQMOffline/Configuration/python/DQMOffline_cff.py:
```diff
@@ -137,3 +138,12 @@
+from DQMOffline.Muon.miniAOD_cff import *                                                                 
+
+muonMiniAOD = cms.Sequence( MuonMiniAOD )
+DQMOfflineMiniAOD = cms.Sequence( miniAODDQMSequence*muonMonitors_miniAOD*muonMiniAOD )
```
replace muonMiniAOD with MuonMiniAOD

deguio commented on DQMOffline/Configuration/python/DQMOffline_cff.py:
```diff
@@ -112,14 +112,15 @@
                                cscSources *
                                muonMonitors
                               )
+DQMOfflineMuon_miniAOD = cms.Sequence( muonMonitors_miniAOD )
```
not needed

deguio commented on DQMOffline/Configuration/python/autoDQM.py:
```diff
@@ -4,11 +4,11 @@
             'muon': ['DQMOfflineMuon',
                      'PostDQMOffline',
                      'DQMHarvestMuon+DQMCertMuon'],
+            'muon_miniAOD': ['DQMOfflineMuon_miniAOD',
```
no need to edit at all autoDQM.py

deguio commented on DQMOffline/Muon/src/MuonMiniAOD.cc:
```diff
@@ -0,0 +1,161 @@
+  else if (WorkingPoint == "softMuons")
+    return muon::isSoftMuon(*muon1,thePrimaryVertex);
+  else{
+    std::cout << "[E]: MuonMiniAOD. Unknown WP, returning false." << std::endl;
```
please make this a logInfo or logWarning

deguio commented on DQMOffline/Muon/src/MuonMiniAOD.cc:
```diff
@@ -0,0 +1,161 @@
+  else if (WorkingPoint == "softMuons")
+    return muon1 -> isSoftMuon(thePrimaryVertex);
+  else{
+    std::cout << "[E]: MuonMiniAOD. Unknown WP, returning false." << std::endl;
```
please make this a logInfo or logWarning

cmsbuild commented:
+1
Tested at: c8b1236b1a4d6ebbc4b660abf3ed4f5f3a77cf7f
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10662/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/b32b4ead0acbc64f84c194e4db21945dc2328458
https://github.com/cms-sw/cmssw/commit/4dc6357c043a05e5f3e6816305bc65465f723533
https://github.com/cms-sw/cmssw/commit/982e9a7dd5952c54b9a453882649f5899d37e3b2
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10662/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10662/git-merge-result

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10662/summary.html

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

sscruz commented:
I have just addressed the comments.

deguio commented on DQMOffline/Configuration/python/DQMOffline_cff.py:
```diff
@@ -137,3 +138,10 @@
 PostDQMOffline = cms.Sequence()
+
+
+DQMOfflineMiniAOD = cms.Sequence( miniAODDQMSequence*muonMonitors_miniAOD*MuonMiniAOD )
```
please remove this sequence and place the muon miniAOD in the PostDQMOfflineMiniAOD

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
please test

deguio commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
-1 
Tested at: 828a97ae6cd810a0cb474404f4ee1a5b40fff036
 When I ran the RelVals I found an error in the following worklfows: 
 9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>
1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>
1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4.log

</pre>
1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>
1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>
25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>
134.911 step1
<pre>DAS Error</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10709/summary.html

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: d7633f8f7e9beb543a1e848e0f7ce9a8a017aa10
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10719/summary.html

deguio commented on DQMOffline/Muon/python/muonMonitors_cff.py:
```diff
@@ -27,14 +22,14 @@
 muonMonitors = cms.Sequence(muonTrackAnalyzers*
                             dtSegmentsMonitor*
-                            cscMonitor*
+#                            cscMonitor*
```
why this is commented out?

deguio commented on DQMOffline/Muon/python/muonMonitors_cff.py:
```diff
@@ -27,14 +22,14 @@
+muonMonitors_miniAOD = cms.Sequence( muonAnalyzer_miniAOD*
+                                     muIsoDQM_seq_miniAOD)
 
 muonMonitorsAndQualityTests = cms.Sequence(muonMonitors*muonQualityTests)
```
is this used?

deguio commented on DQMOffline/Muon/python/muonMonitors_cff.py:
```diff
@@ -27,14 +22,14 @@
-if eras.phase1Pixel.isChosen() :
-    muonMonitors.remove(muonAnalyzer)
+
+muonMonitors_miniAOD = cms.Sequence( muonAnalyzer_miniAOD*
```
is this used?

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

sscruz commented:
muonMonitors_miniAOD is indeed needed. It was originally included in the sequence DQMOfflineMiniAOD, which I was asked to erase. I have moved it into PostDQMOfflineMiniAOD.

sscruz commented:
muonMonitors_miniAOD is indeed needed. It was originally included in the sequence DQMOfflineMiniAOD, which I was asked to erase. I have moved it into PostDQMOfflineMiniAOD.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

deguio commented:
+1
provided it passes the tests

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: 0d6ba51c810d773665a48efc0d3e82bf5e4ed009
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10874/summary.html

davidlange6 commented on DQMOffline/Muon/src/MuonMiniAOD.cc:
```diff
@@ -0,0 +1,161 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <string>
+#include "TMath.h"
```
is this include needed?

davidlange6 commented on DQMOffline/Muon/src/DiMuonHistograms.cc:
```diff
@@ -180,29 +175,30 @@ void DiMuonHistograms::analyze(const edm::Event & iEvent,const edm::EventSetup&
+	    for (unsigned int iEtaRegion=0; iEtaRegion<3; iEtaRegion++){
+	      if (iEtaRegion==0) {EtaCutMin= 0.;         EtaCutMax=2.4;       }
+	      if (iEtaRegion==1) {EtaCutMin= etaBMin;    EtaCutMax=etaBMax;   } 
+	      if (iEtaRegion==2) {EtaCutMin= etaECMin;   EtaCutMax=etaECMax;  }
```
could these if statements go into a place where they get executed once? (e.g., just store them into an array that you then loop over)
@sscruz

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10874/summary.html

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 1f752480806ea68e6fee4397bb82de36baf2fe76
 I found an error when building: 
 
 <pre>>> Subsystem DQMOffline built
--- Registered EDM Plugin: DQMServicesExamplesPlugins
>> Leaving Package DQMServices/Examples
>> Package DQMServices/Examples built
gmake: Target `all' not remade because of errors.
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10942/summary.html

sscruz commented:
Hi, I believe this error is not related to this PR. Can someone confirm this?

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 1f752480806ea68e6fee4397bb82de36baf2fe76
 When I ran the RelVals I found an error in the following worklfows: 
 9.0 step3
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step3_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>
140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>
134.911 step3
<pre>runTheMatrix-results/134.911_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT.log

</pre>
25.0 step3
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>
1306.0 step3
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step3_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>
1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4.log

</pre>
50202.0 step3
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step3_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>
25202.0 step3
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step3_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/10952/summary.html

deguio commented on DQMOffline/Muon/src/DiMuonHistograms.cc:
```diff
@@ -67,42 +68,33 @@ void DiMuonHistograms::bookHistograms(DQMStore::IBooker & ibooker,
 
-  int nBin = 0, nBinLM = 0;
+  int nBin[3] = {etaBin,etaBBin,etaEBin};
+  EtaName[0]  = ""; EtaName[1] = "_Barrel"; EtaName[3] = "_EndCap";
```
EtaName[3] = "_EndCap";
should be
EtaName[2] = "_EndCap";

deguio commented:
the last commit is buggy..  :(
F.

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

folguera commented:
sorry for the mess there... we hope now it is fixed and satisfying all the comments from you.

deguio commented:
ciao,
I believe that yes, but the branch is nor mergeable anymore. could you please rebase?
thanks,
F.

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
still not mergeable

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

folguera commented:
@deguio I think we fixed the issue, could you please trigger the tests?  Thanks and sorry for the inconvenience...

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: c29aa24298cfa75c85207dc436ea605a65ca63d2
 I found an error when building: 
 
 <pre>Copying tmp/slc6_amd64_gcc493/src/DQMServices/Components/bin/fastHadd/fastHadd to productstore area:
>> Building edm plugin tmp/slc6_amd64_gcc493/src/DQMServices/FwkIO/test/DQMServicesFwkIOTestPlugins/libDQMServicesFwkIOTestPlugins.so
Copying tmp/slc6_amd64_gcc493/src/DQMServices/ClientConfig/src/DQMServicesClientConfig/libDQMServicesClientConfig.so to productstore area:
/afs/cern.ch/cms/sw/ReleaseCandidates/vol1/slc6_amd64_gcc493/cms/cmssw-patch/CMSSW_8_0_X_2016-02-07-2300/external/slc6_amd64_gcc493/lib/libssl.so: error adding symbols: Bad address
>> Building edm plugin tmp/slc6_amd64_gcc493/src/DQMServices/FwkIO/plugins/DQMServicesFwkIOPlugins/libDQMServicesFwkIOPlugins.so
collect2: error: ld returned 1 exit status
>> Deleted: tmp/slc6_amd64_gcc493/src/DQMServices/FwkIO/test/TestDQMServicesFwkIOScripts/TestDQMServicesFwkIOScripts
gmake: *** [tmp/slc6_amd64_gcc493/src/DQMServices/FwkIO/test/TestDQMServicesFwkIOScripts/TestDQMServicesFwkIOScripts] Error 1
>> Building shared library tmp/slc6_amd64_gcc493/src/DQMServices/Diagnostic/src/DQMServicesDiagnostic/libDQMServicesDiagnostic.so
>> Building edm plugin tmp/slc6_amd64_gcc493/src/DQMServices/StreamerIO/test/DQMServicesStreamerIOTestPlugins/libDQMServicesStreamerIOTestPlugins.so
Leaving library rule at DQMServices/ClientConfig
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/11072/summary.html

sscruz commented:
@deguio I think that the compilation error is not related to this PR.

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: c29aa24298cfa75c85207dc436ea605a65ca63d2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/11098/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/11098/summary.html

deguio commented:
I think there is still a comment by david to be addressed, but the code looks in good shape to me.

folguera commented:
Sorry for this, do you prefer that we address this comment now or in a different PR?

sscruz commented:
Hi,  I've been working on completely addressing David's comment. The code that I've produced compiles correctly.  However, I've been finding some errors when trying to test it, not related to the modules that I'm modifying, but that don't allow me to test them. When running  [1], I get [2]. Any hint in which can be the cause? I'm using CMSSW_8_0_0_pre5. 
Cheers and many thanks.

[1]  cmsDriver.py test -s DQM:@miniAODDQM -n 30 --eventcontent DQM --conditions auto:com10 --geometry Ideal --mc --filein /store/relval/CMSSW_8_0_0_pre5/RelValZMM_13/MINIAODSIM/PUpmx25ns_80X_mcRun2_asymptotic_v1-v1/00000/D0E4D3A1-2CC6-E511-B5D6-0025905A60EE.root

[2] DQM:@miniAODDQM,ENDJOB
entry /store/relval/CMSSW_8_0_0_pre5/RelValZMM_13/MINIAODSIM/PUpmx25ns_80X_mcRun2_asymptotic_v1-v1/00000/D0E4D3A1-2CC6-E511-B5D6-0025905A60EE.root
 ################################################################### 
 # WARNING: this module is deprecated.                             # 
 # Please use CondCore.CondDB.CondDB_cfi.py                        # 
 ################################################################### 
Step: DQM Spec: ['@miniAODDQM']
Step: ENDJOB Spec: 
# Conditions read from  CMS_CONDITIONS  via FrontierProd 
Starting  cmsRun  test_DQM.py
 ################################################################### 
 # WARNING: this module is deprecated.                             # 
 # Please use CondCore.CondDB.CondDB_cfi.py                        # 
 ################################################################### 
# Conditions read from  CMS_CONDITIONS  via FrontierProd 
----- Begin Fatal Exception 10-Feb-2016 18:15:34 CET-----------------------
An exception of category 'PluginLibraryLoadError' occurred while
   [0] Constructing the EventProcessor
Exception Message:
unable to load /afs/cern.ch/work/s/sesanche/private/DQM/Feb10/CMSSW_8_0_0_pre6/lib/slc6_amd64_gcc530/poisoned/plugin-poisoned-L1TriggerL1TCalorimeterPlugins.so because /afs/cern.ch/work/s/sesanche/private/DQM/Feb10/CMSSW_8_0_0_pre6/lib/slc6_amd64_gcc530/poisoned/plugin-poisoned-L1TriggerL1TCalorimeterPlugins.so: cannot open shared object file: No such file or directory
----- End Fatal Exception -------------------------------------------------

deguio commented:
the errors don't seem to be related. could you try with a more recent IB as a starting point?

sscruz commented:
Hi, I've tried with a couple of IBs: CMSSW_8_0_X_2016-02-01-2300, CMSSW_8_0_X_2016-02-11-2300, and also with CMSSW_8_0_0_pre5, but all of them deliver a similar error.

deguio commented:
I don't see errors in the last automatic test

deguio commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

deguio commented:
forgot to mention that a last rebase is needed apparently.
thanks!
F.

cmsbuild commented:
Pull request #12996 was updated. @cmsbuild, @vanbesien, @deguio, @davidlange6 can you please check and sign again.

deguio commented:
+1

deguio commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_8_0_X IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @slava77, @davidlange6, @degano, @smuzaffar

cmsbuild commented:
+1
Tested at: 4b22e38f37d5d97b686dfc9fc46f824c39c92f38
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/11347/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-12996/11347/summary.html

DirtyDan88 committed:
PCL-DQM-plots: implemented mp-file reader and DQMEDAnalyzer for histo booking and filling

DirtyDan88 committed:
fixed some typos

DirtyDan88 committed:
added config for MillePedeFileReader

DirtyDan88 committed:
removed unused includes

VinInn committed:
add index for gdets

VinInn committed:
faster, not as much as expected

VinInn committed:
infinitely faster hash

VinInn committed:
cure expensive math

VinInn committed:
ofast

VinInn committed:
exchange tests: slower second

VinInn committed:
speed up EtaPhiMeasurementEstimator

kpedro88 committed:
remove unnecessary include of HcalDbHardcode

kpedro88 committed:
make HcalHardcodeCalibrations and HcalDbHardcode configurable for simple parameters

VinInn committed:
modernize EtaPhiRange, remove obsolere GlobalDetRange

VinInn committed:
devirtualize, speed up math

VinInn committed:
make atan2 3bits more precise

VinInn committed:
fix trivial bug, no regression

VinInn committed:
modernize and add Ofast where relevant

VinInn committed:
leave both versions, use old one

VinInn committed:
use std::shared_ptr

VinInn committed:
change from map to heap

VinInn committed:
use priority_queue

VinInn committed:
remove debug

VinInn committed:
make DynArray std compatible

VinInn committed:
test swa[

VinInn committed:
remove debug

VinInn committed:
dummy out hit etaphi estimator

gregor-mittag committed:
Adaptions to the unified track selection sequence.

 - put into actual use for the first time

gregor-mittag committed:
Make the unified track selection applicable to 'generalTracks'.

 - treat them like MinBias tracks
 - replaced the check for string identity with checks for string equality
   -> not sure if needed here, but can be different in general and leads to
      strange errors in such cases

 TODOS:
  - do not copy monitoring files for the individual jobs to afs, but only to eos

gregor-mittag committed:
Replace usage of condDBv1 tools with condDBv2 in pede script template.

gregor-mittag committed:
Replace aligncond_split_iov in pede script with script using condDBv2 tools.

VinInn committed:
return components by const reference

gregor-mittag committed:
Remove CASTOR prestage block in MillePede script creation process.

kpedro88 committed:
remove very old unused hcal db code

kpedro88 committed:
update hardcode parameters

VinInn committed:
add wrapper in other places

VinInn committed:
remove debug

kpedro88 committed:
useHFUpgrade overrides testHFQIE10

kpedro88 committed:
make Hcal Phase 0.5 customization

Commit:
Update the ganging for BH as discussed in the simulation meeting

Commit:
Resolve conflicts

Commit:
Correct one more xml file

Commit:
Correct one more py file

davidlange6 committed:
Merge pull request #13870 from smuzaffar/testRandomNumberGenerators-cleanup

remove unused testRandomNumberGenerators binary

cmsbuild committed:
Merge pull request #13857 from jruizvar/addExoPaths81X

Adding double-electron triggers to EXO DQM (81X)

Sam-Harper committed:
removing jetcoreregional tracks from tracker isolation sum

argiro committed:
Added edtoken to EcalDumpRaw

jasperlauwers committed:
Adding new emu path

Commit:
EnableGEMMeasurement part of FilterParameters

mulhearn committed:
complete migration from static variable in OMTF configuration

mulhearn committed:
make static config member variable const in BMTF config

mulhearn committed:
only consume things we are actually going to eat

Commit:
remove redundant subsystems

Commit:
replace isChosen by customization functions

Commit:
bugfix

Commit:
Add missing GEM/ME0 pieces to mixing module

kpedro88 committed:
skip DIGI2RAW and RAW2DIGI for hcal phase 0.5

Commit:
Merged refs/pull/13880/head from repository cms-sw

Commit:
GEM/ME0 era customizations for validation

Commit:
clean-up

Commit:
clean-up

Commit:
Add missing .dat files to make GEM local RECO work in Phase2

davidlange6 committed:
Merge pull request #13768 from cms-l1t-offline/pr-l1t-tsg-v4-81x

Bring L1T emulation up to date with l1t-tsg-v4 tag

makortel committed:
Fix compilation under EDM_ML_DEBUG

VinInn committed:
remove debug, add comments

VinInn committed:
remove cout, add comment

rafaellopesdesa committed:
Put the header files for MVA electron selection in VID in the right place

rafaellopesdesa committed:
delete old headers in wrong place - MVA electron id in VID

gregor-mittag committed:
Fixed an old bug in 'mps_setupm.pl'.

 - when duplicating the pede job folder before the mille jobs are done the
   resulting pede job script is wrong
   -> this is not always intended and an erroneous check meant to prevent this
      is fixed by this commit

Commit:
Scouting triggers for 2016

Commit:
New DisplacedJet Paths

Dr15Jones committed:
Fix evaluation of function arguments for FormulaEvaluator

The FormulaEvaluator was not using the proper evaluator for the argument
passed to a function if the argument was a complex expression.
Added unit tests for the failing cases.

wmtan committed:
Use std::shared_ptr and std::make_shared

wmtan committed:
Use std::shared_ptr and std::make_shared

rafaellopesdesa committed:
same modification for photon MVA

wmtan committed:
use std::shared_ptr and std::make_shared

Commit:
delete redundancy

wmtan committed:
use std::shared_ptr and std::make_shared

Commit:
delete redundant customs

wmtan committed:
use std::shared_ptr and std::make_shared

davidlange6 committed:
Merge pull request #13885 from makortel/fixDebug

Fix compilation under EDM_ML_DEBUG

davidlange6 committed:
Merge pull request #13873 from Sam-Harper/HEEPTrkIsoFix81X

Bug fix for Electron Tracker Isolation used by HEEP

davidlange6 committed:
Merge pull request #13872 from bsunanda/Phase2-hgx47

bsunanda:Phase2-hgx47 Update the ganging for BH as discussed in the simulation meeting

cmsbuild committed:
Merge pull request #13834 from bsunanda/Phase2-gem11

bsunanda:Phase2-gem11 More realistic ME0 geometry and extended GE21 geometry

davidlange6 committed:
Merge pull request #13867 from davidlange6/phase2sim

add some phase2 (sim only) workflows to the default matrix

davidlange6 committed:
Merge pull request #13868 from akhukhun/fix-hf-lut-crash-81X

fix HF TP transcoder crash.

rekovic committed:
Merged l1t-dqm-CMSSW_8_1_0_pre1 from repository cms-l1t-offline

Conflicts:
	DataFormats/L1TMuon/interface/EMTF/SP.h
	EventFilter/L1TRawToDigi/src/implementations_stage2/EMTFBlockSP.cc

mmusich committed:
Updating L1 Menu and HLT JECs

wmtan committed:
use std::shared_ptr and std::make_shared

wmtan committed:
use std::make_shared

davidlange6 committed:
Merged refs/pull/13892/head from repository cms-sw

davidlange6 committed:
Merged refs/pull/13863/head from repository cms-sw

rekovic committed:
External conditions (technical) triggers for L1 Stage2. Include in simulation sequence.

kpedro88 committed:
fix typo in customization

mzarucki committed:
Updated L1TStage2uGT DQM module

ianna committed:
* Remove obsolete customizations

swang373 committed:
Merge pull request #227 from mzarucki/uGT_DQM_v3_81X

Updated L1TStage2uGT DQM module

swang373 committed:
Fix global tag loading for testing DQM offline

The file "FrontierCondition_GT_Offline_cfi.py" no longer exists in
DQM/Integration/python/config/. The DQM sequence now loads the file
"FrontierCondition_GT_cfi.py" when testing offline as well.

cerminar committed:
cleanup and go through the mpreader only if writing conditions to DB is required

cerminar committed:
add the parameters for the millepedefilereader

mbandrews committed:
Add TimingTask sequence. Consolidate TimingClient time windows.

mbandrews committed:
Add TTF plots. Remove subdet check for masking status by SM.

Commit:
Remove wrong warnings

Commit:
Correct forward muon part for 3 2023 scenario

Commit:
Update the GE11 chambers as well

lgray committed:
address code review

VinInn committed:
use heap and local arrays in seed finder

davidlange6 committed:
move to 2016 era and 2016 hlt where it is working

davidlange6 committed:
Merge pull request #13897 from wmtan/StdSharedPtrInGeometry

use std::shared_ptr and std::make_shared (EventSetup in Geometry/Alignment/Visualization)

davidlange6 committed:
Merge pull request #13896 from wmtan/StdSharedPtrInFastSim

Use std::shared_ptr and std::make_shared (EventSetup in FastSim)

davidlange6 committed:
Merge pull request #13866 from smuzaffar/L1Trigger-CSCTrackFinder-to-external

Move L1Trigger/CSCTrackFinder/src/core_ files to external package

davidlange6 committed:
Merge pull request #13831 from VinInn/KullbackLeibler

Speedup Gsf component merging, return components by const reference

davidlange6 committed:
Merge pull request #13819 from VinInn/SpeedHLT81

Speed seeding for HLT

cmsbuild committed:
Merge pull request #13903 from cms-l1t-offline/hlt-l1tseed-update-80x

HLT L1T interface updates - trhow exceptions and L1GlobalDecision  80x

Commit:
HMDQM

davidlange6 committed:
fix typo

davidlange6 committed:
Merge pull request #13901 from wmtan/UseMakeShared

use std::make_shared in framework (again)

davidlange6 committed:
Merge pull request #13919 from davidlange6/hltfor81xrelval

move to 2016 era and 2016 hlt where it is working in relvals

cmsbuild committed:
Merge pull request #13899 from wmtan/StdSharedPtrInFullsim

Use std::shared_ptr and std::make_shared (EventSetup in Simulation)

cmsbuild committed:
Merge pull request #13898 from wmtan/StdSharedPtrInReco

use std::shared_ptr and std::make_shared (EventSetup in Reco)

Commit:
Clean where address is used as a pointer

Address is not a pointer, thus comparing it to a 0/NULL/nullptr does not
make sense -- it's always false. Compiler can assume that address is
never 0/NULL/nullptr.

Tested with GCC 6.0.0 (r234661):

    DQM/CSCMonitorModule/plugins/CSCDQM_EventProcessor_processCSC.cc:87:15: error: the compiler can assume that the address of 'data' will never be NULL [-Werror=address]
    DQMOffline/JetMET/src/HTMHTAnalyzer.cc:85:21: error: the compiler can assume that the address of 'triggerResults' will always evaluate to 'true' [-Werror=address]
    DQMOffline/PFTau/src/Benchmark.cc:25:10: error: the compiler can assume that the address of 'b' will always evaluate to 'true' [-Werror=address]
    DQMOffline/PFTau/src/Benchmark.cc:35:10: error: the compiler can assume that the address of 'b' will always evaluate to 'true' [-Werror=address]
    DQMOffline/PFTau/src/Benchmark.cc:45:10: error: the compiler can assume that the address of 'b' will always evaluate to 'true' [-Werror=address]
    DQMOffline/PFTau/src/Benchmark.cc:63:10: error: the compiler can assume that the address of 'b' will always evaluate to 'true' [-Werror=address]
    DQMOffline/PFTau/src/Benchmark.cc:80:10: error: the compiler can assume that the address of 'b' will always evaluate to 'true' [-Werror=address]
    DQM/Physics/src/EwkMuLumiMonitorDQM.cc:359:14: error: the compiler can assume that the address of 'toc' will never be NULL [-Werror=address]
    DQM/SiStripCommissioningSummary/src/DaqScopeModeSummaryFactory.cc:91:24: error: the compiler can assume that the address of 'summary_histo' will always evaluate to 'true' [-Werror=address]
    DQM/SiStripCommissioningSummary/src/FedTimingSummaryFactory.cc:108:24: error: the compiler can assume that the address of 'summary_histo' will always evaluate to 'true' [-Werror=address]
    DQM/SiStripCommissioningSummary/src/SummaryHistogramFactory.cc:103:8: error: the compiler can assume that the address of 'summary_histo' will always evaluate to 'true' [-Werror=address]
    DQM/SiStripCommissioningSummary/src/SummaryPlotFactoryBase.cc:131:24: error: the compiler can assume that the address of 'summary_histo' will always evaluate to 'true' [-Werror=address]

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Add missing <cmath> include; use std::fabs

Tested with GCC 6.0.0 (r234661):

    DQMOffline/JetMET/interface/SusyDQM/HT.h:38:42: error: call of overloaded 'abs(double)' is ambiguous

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

VinInn committed:
do not use Ref...

lgray committed:
use vdt in PFRecHit

VinInn committed:
remove constructors, remove local hit

VinInn committed:
Merged speed_pfrechit from repository lgray

VinInn committed:
Merged SpeedPFC81 from repository VinInn

argiro committed:
moved to stream interface

dsrankin committed:
adding new b2g dijet control paths to config

VinInn committed:
reserve shrink

davidlange6 committed:
more era migration

cmsbuild committed:
Merge pull request #13913 from ianna/config-cleanup

Remove Obsolete Customizations

davidlange6 committed:
Merge pull request #13895 from wmtan/StdSharedPtrInCalibration

use std::shared_ptr and std::make_shared (EventSetup in Calibration)

cmsbuild committed:
Merge pull request #13891 from cms-l1t-offline/l1t-dqm-CMSSW_8_1_X_2016-04-01

L1Trigger DQM Combo PR 13800 fixed merge conflict

VinInn committed:
try clever reserve, does not speed much

cerminar committed:
add cff for DQM module

cerminar committed:
add DQM module to the sequence

mdhildreth committed:
Add eras switch for L1 configuration for PreMixing

davidlange6 committed:
Merge pull request #13941 from davidlange6/hlt81xv2

more era migration in relval matrix

davidlange6 committed:
Merge pull request #13939 from mdhildreth/PreMixing_Stage2L1Fix

PreMixing era switch for Stage 2 L1 configuration

vanbesien committed:
Removed old and obsolute fedtest config

To explain: For a long time (5 years) we had two configurations running:
FED and FEDTest. At was not really documented and nobody really knew
what it was. It seemed that FED was broken an FEDTest was showing
useful stuff. So recently we rectified this situation by making FED the
real working and fixed one, meaning that FEDTest should stop existing
and should be removed.

jeongeun committed:
specify module

specific module

specific module

specific module

 specific module

VinInn committed:
igve up with move constructor, use emplace_back and pop

cerminar committed:
remove problematic fillDescriptions and remove creation of canvas

cerminar committed:
clone the pset for the millepede file reader

VinInn committed:
fix SpikeCleaner as well

Commit:
modifiy MillePedeDQMModule to store information in a single histogram

davidlange6 committed:
Merge pull request #13894 from wmtan/StdSharedPtrInAnalysis

use std::shared_ptr and std::make_shared (EventSetup in Analysis)

cmsbuild committed:
Merge pull request #13931 from drankincms/b2gdqm-dijet-control

Adding new b2g dijet control paths to B2G HLT DQM

cmsbuild committed:
Merge pull request #13929 from zc11/DQM2016HM

HLT DQM for High Multiplicity Triggers for 2016 low PU run for 81X

cmsbuild committed:
Merge pull request #13918 from mbandrews/Timing+TTF81X

Add TimingTask sequence. Add TTF plots.

cmsbuild committed:
Merge pull request #13916 from bsunanda/Phase2-gem13

bsunanda:Phase2-gem13 Correct forward muon part for 3 2023 scenario

davidlange6 committed:
Merge pull request #13726 from archiron/ElectronValidation_80X_HistoName

Electron validation 80X histo name modification

davidlange6 committed:
Merge pull request #13887 from jruizvar/newExoPaths81X

Add triggers to EXO DQM (81X)

davidlange6 committed:
Merge pull request #13876 from argiro/ecaldumpraw_edtoken

Added edtoken to EcalDumpRaw

davidlange6 committed:
Merge pull request #13927 from davidlt/gcc-600-r234661-dqm-cleanup

DQM GCC 6.0.0 (r234661) cleanup (address used as a pointer, std::fabs)

davidlange6 committed:
Merge pull request #13878 from jasperlauwers/81X_HLTriggerOffline_Higgs_MuElpath

Adding new emu path

davidlange6 committed:
Merge pull request #13851 from popovvp/dqm81X

add summary report

cmsbuild committed:
Merge pull request #13835 from lgray/reworked_rechit_scheme

HGCal Reworked rechit scheme

Commit:
take out empty .dat files

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

VinInn committed:
more noexcept

davidlange6 committed:
Merge pull request #13880 from dildick/for-GEM-digi-eras-customization

SIM-DIGI-RECO-Validation era based customizations for GEM/ME0

davidlange6 committed:
Merge pull request #13947 from vanbesien/removing_old_and_obsolute_fedtest_config_81X

Removed old and obsolete fedtest config 81X

kpedro88 committed:
simpler presample assignment for QIE10

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

davidlange6 committed:
Merge pull request #13832 from kpedro88/Phase1-HF5

QIE10 bug fixes

wmtan committed:
Use std::shared_ptr in EventSetup interface

silviodonato committed:
add DQM code for Higgs

silviodonato committed:
Higgs DQM fix

silviodonato committed:
fix offline DQM

silviodonato committed:
fix online DQM

silviodonato committed:
fix in DQM online

kpedro88 committed:
code review for HcalDbHardcode

VinInn committed:
use tables non log/pow, remove reduntant checks in topologies

VinInn committed:
Merged SpeedPFC81 from repository VinInn

VinInn committed:
import mods by had

VinInn committed:
Merged SpeedES81 from repository VinInn

davidlange6 committed:
Revert "Move L1Trigger/CSCTrackFinder/src/core_ files to external package"

davidlange6 committed:
Merge pull request #13974 from cms-sw/revert-13866-L1Trigger-CSCTrackFinder-to-external

Revert "Move L1Trigger/CSCTrackFinder/src/core_ files to external package"

cerminar committed:
move to DQMEDHarvester interface

schneiml committed:
Read number of blades from tracker

schneiml committed:
Also retrieve the ladder counts, for consistency.

davidlange6 committed:
Revert "bsunanda:Phase2-gem13 Correct forward muon part for 3 2023 scenario"

davidlange6 committed:
Merge pull request #13985 from cms-sw/revert-13916-Phase2-gem13

Revert "bsunanda:Phase2-gem13 Correct forward muon part for 3 2023 scenario"

cmsbuild committed:
Merge pull request #13660 from mtosi/from-CMSSW_8_1_X_2016-03-08-2300_update_messageLog

update monitoring of tracking message loggger

davidlange6 committed:
Merge pull request #13890 from Dr15Jones/fixArgEvaluationInFormulaEvaluator

Fix evaluation of function arguments for FormulaEvaluator

mmusich committed:
Fixed payloads for JEC at HLT, adding new L1 menu to cosmics workflow and introducing uncertainties for 50ns JEC

davidlange6 committed:
Merge pull request #13965 from wmtan/StdSharedPtrInL1

Use std::shared_ptr in EventSetup interface (L1 and DB)

cmsbuild committed:
Merge pull request #13911 from rafaellopesdesa/plugin_fix_VID

VID MVA headers in plugin directory - 81X version

cmsbuild committed:
Merge pull request #13990 from mmusich/GT_81X_HLTJEC

Fixed payloads for JEC at HLT, adding new L1 menu to cosmics workflow and introducing uncertainties for 50ns JEC

Commit:
Correct the shortcoming in PR13916

kpedro88 committed:
fix QIE10 shape: bins start at -0.5

wmtan committed:
use unique_ptr, not auto_ptr, in EventSetup

smuzaffar committed:
re-enable testCommonToolsUtil unit tests

few tests were disabled as those either needed input or were hang due to some other reason.

VinInn committed:
add abs

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

cmsbuild committed:
Merge pull request #13915 from bsunanda/Run2-hcx76

bsunanda:Run2-hcx76 Remove wrong warnings

Commit:
Initialize lastModuleIndex to 0

While compiling with GCC 6.0.0 (r234771) in C++1z mode compiler produces
extra diagnostic:

    FWCore/Framework/src/Schedule.cc:1372:43: error: 'lastModuleIndex'
may be used uninitialized in this function [-Werror=maybe-uninitialized]

Initialize lastModuleIndex to 0.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Be explicit about 'data' usage

While compiling with GCC 6.0.0 (r234771) in C++1z mode compiler
produces extra diagnotic messages:

    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:47:3: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:103:30: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:104:27: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:110:31: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:111:5: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:116:5: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:121:5: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:137:26: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:138:7: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:230:4: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:239:6: error: reference to 'data' is ambiguous
    L1Trigger/GlobalCaloTrigger/test/testElectronSorter.cpp:250:16: error: reference to 'data' is ambiguous
    RecoVertex/BeamSpotProducer/test/global.h:35:25: error: template argument 1 is invalid
    RecoVertex/BeamSpotProducer/test/global.h:35:25: error: template argument 2 is invalid
    RecoVertex/BeamSpotProducer/test/global.h:36:31: error: expected initializer before 'zDataConstIter'
    RecoVertex/BeamSpotProducer/test/global.h:37:25: error: expected initializer before 'zDataIter'

Avoid using plain 'data', because in C++1z (C++17) there is a new
std::data() function and if using namespace std exist it could
result in unexpected result visible in above diagnotic messages.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

mkirsano committed:
Correct secondary plugin and an example in LHEInterface

VinInn committed:
avoid memory churn in prune

cmsbuild committed:
Merge pull request #14000 from cms-sw/smuzaffar-patch-2

re-enable testCommonToolsUtil unit tests

wmtan committed:
OwnVector should use unique_ptr, not auto_ptr

wmtford committed:
Add consumes support for TestAssociator

bendavid committed:
fix missing file close, and add protection for maximum random seed needed for madgraph

davidlange6 committed:
Merge pull request #13999 from wmtan/UseUniquePtrInEventSetup

use unique_ptr, not auto_ptr, in EventSetup

lgray committed:
Fix out-of-bounds bug in HGCFEElectronics

Logic was checking bounds with > instead of >=

VinInn committed:
Update BuildFile.xml

use only header. do not overload std math

VinInn committed:
Merged SpeedPFC81 from repository VinInn

cmsbuild committed:
Merge pull request #13966 from silviodonato/DQMVBF

Change of VBF Hbb HLT path names in DQM code (online,offline,relval)

cmsbuild committed:
Merge pull request #14003 from mkirsano/lheinterface_fix1

Correct secondary plugin and an example in LHEInterface

cmsbuild committed:
Merge pull request #13996 from bsunanda/Phase2-gem14

bsunanda:Phase2-gem14 Update the forward muon definition in 2023 Geometry scenario

cmsbuild committed:
Merge pull request #13986 from schneiml/validation-nblades-fix

SiPixelDigiValid::analyse: blade number fix

davidlange6 committed:
Merge pull request #14012 from lgray/patch-1

Fix out-of-bounds bug in HGCFEElectronics

wmtan committed:
Remove EventSetup support for boost::shared_ptr and std::auto_ptr

rekovic committed:
Use vector to locally store extern decisions and loop over it.

gartung committed:
The return values of gSystem->Load(library) can be 0 or 1.

VinInn committed:
Update PFRecHit.cc

allow clang and icc to compile it.
if may become utterly slow

avetisya committed:
Finalizing JSON monitoring for 2016

davidlange6 committed:
Merge pull request #14018 from wmtan/EventSetupNoDeprecatedPtrSupport

Remove EventSetup support for boost::shared_ptr and std::auto_ptr

cmsbuild committed:
Merge pull request #14006 from wmtford/testAssociator_w_consumes

Add consumes support for TestAssociator

cmsbuild committed:
Merge pull request #14005 from bendavid/lhescanfix_81x

fixes for parameter scan functionality

cmsbuild committed:
Merge pull request #14001 from davidlt/gcc6-fw-schedule-init

Initialize lastModuleIndex to 0

cmsbuild committed:
Merge pull request #13957 from ghellwig/new-MillePede-production-system_81X

New MillePede production system

davidlange6 committed:
Merge pull request #14009 from wmtan/UseUniquePtrInOwnVector

OwnVector should use unique_ptr, not auto_ptr

davidlange6 committed:
Merge pull request #14002 from davidlt/gcc6-cxx1z-std-data

Be explicit about 'data' usage

davidlange6 committed:
Merge pull request #13997 from kpedro88/Phase1-HF6

fix QIE10 shape: bins start at -0.5

VinInn committed:
Update LocalMaximumSeedFinder.cc

VinInn committed:
Update CorrectedECALPFClusterProducer.cc

remove cout

VinInn committed:
Merged SpeedPFC81 from repository VinInn

VinInn committed:
missing the MessageLogger, the only file in the whole CMSSW

gartung committed:
fix two other scripts that check the return of gSystem->Load()

cmsbuild committed:
Merge pull request #13371 from lgray/hgc_trigger_fwk_80X

HGCal L1 Trigger Framework

deguio commented:
need to have 81x merged first

Dr15Jones committed:
Removed use of assert(nullptr)

Using gcc 5.3 on OS X failed to compile because nullptr could not auto convert to bool in the assert expression. Given the expressing does not make much sense, it was replaced with assert(false).

cerminar committed:
cleanup

cmsbuild committed:
Merge pull request #13826 from dmitrijus/bril_pr_80x

Implement BRIL DQM client

deguio committed:
fix EvD auto GT following the evolution of the T0 API

smuzaffar committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

davidlange6 committed:
Merge pull request #14037 from Dr15Jones/remove_assert_nullptr

Removed use of assert(nullptr)

cmsbuild committed:
Merge pull request #14031 from avetisya/json-mon-final-stage2-adaptation

Finalizing JSON monitoring for 2016

cmsbuild committed:
Merge pull request #13948 from jeongeun/smallfixModule81X

small update of specifying event contents modules 81X

cmsbuild committed:
Merge pull request #13905 from cms-l1t-offline/extcond-trig-l1t-80x

External conditions (technical) triggers for L1 Stage2.

cmsbuild committed:
Merge pull request #14042 from deguio/fixEvD

fix EvD config following the evolution of the T0 API

cmsbuild committed:
Merge pull request #13983 from cerminar/PCLDQMPlots_81X_v0

PCL SiPixel Alignment: integration of quality checks on payload and DQM module

cmsbuild committed:
Merge pull request #14022 from gartung/edm-Check-Class-Version-fix

The return values of gSystem->Load(library) can be 0 or 1.

davidlange6 committed:
Merge pull request #13959 from VinInn/SpeedPFC81

SpeedUp ParticleFlow for HLT

cmsbuild committed:
Merge pull request #13861 from kpedro88/Phase1-DbHardcode1

Configurable hardcode conditions for HCAL

Commit:
Fix conflicts.

cmsbuild commented:
Pull request #12996 was updated. @perrotta, @cmsbuild, @cvuosalo, @fwyzard, @cerminar, @Martin-Grunewald, @rekovic, @franzoni, @deguio, @slava77, @ggovi, @mmusich, @vanbesien, @mulhearn, @davidlange6 can you please check and sign again.

slava77 commented:
-1
looks like the PR is based on 81X, while it is instead submitted to 80X

sscruz commented:
Hi, sorry for the inconvenience. I think I misunderstood @deguio 's last comment. Do I have to rebase for 80X or 81X?

slava77 commented:
https://github.com/cms-sw/cmssw/pull/12996#issuecomment-208940479
said ``` need to have 81x merged first ```
nothing about a rebase, all new feature PRs should be merged to 81X (development release) first and then proposed to merge to a production release like 80X.

deguio commented:
-1

davidlange6 commented:
i'm closing this - lets sort it out in 81x.
